samples/nrf9160/modem_shell/src/gnss/gnss_shell.c

```diff
@@ -81,32 +51,17 @@ static int cmd_gnss_delete(const struct shell *shell, size_t argc, char **argv)
 
 static int cmd_gnss_delete_ephe(const struct shell *shell, size_t argc, char **argv)
 {
-	if (gnss_running) {
-		mosh_error("%s: stop GNSS to execute command", argv[0]);
-		return -ENOEXEC;
-	}
-
-	return gnss_delete_data(GNSS_DATA_DELETE_EPHEMERIDES);
+	return gnss_delete_data(GNSS_DATA_DELETE_EPHEMERIDES) == 0 ? 0 : -ENOEXEC;
```

Comment:
What happens if you just return what `gnss_delete_data` returns?

Reply:
I'm just looking that we will miss the real error code returned by the functions but I guess there are logs elsewhere which can be used if needed. But you know whether the real error code is relevant especially hen calling a more complex function than `gnss_delete_data`.

Reply:
The idea of the change is, that possible error is printed already in gnss.c. In gnss_shell.c we can just return a proper return value to the shell.